src/result/mod.rs

```diff
@@ -26,81 +21,51 @@ mod status;
 pub use self::status::Status;
 
 /// Return type of most UEFI functions. Both success and error payloads are optional.
```

Comment:
Should we mention here something about how warnings are treated as errors?

Reply:
Good call, I've overhauled the documentation here a bit. The module wasn't public so I dropped most of the module docstring and expanded the `Result` docstring instead, since that's probably where it's most useful anyway.